src/Board/List/TasksCards/TasksCard.js

```diff
@@ -88,8 +88,9 @@ let TasksCards = () => {
       const items = reorder(tasksFilter[sInd], source.index, destination.index);
       const newState = [...tasksFilter];
       newState[sInd] = items;
-
-      dispatch(setTaskState(newState, listsId));
+      let reducerState = [];
+      newState.map((item) => reducerState.push(...item));
```

Comment:
map should be always pure function.

If you want to flat array you could use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat